src/babel-serializer/geometry/stl.cc

```diff
@@ -0,0 +1,272 @@
+#include "stl.hh"
+
+#include <cstring> // std::memcpy
+#include <limits>
+
+#include <clean-core/from_string.hh>
+#include <clean-core/typedefs.hh>
+
+babel::stl::geometry babel::stl::read(cc::span<const std::byte> data, babel::stl::read_config const& cfg, babel::error_handler on_error)
+{
+    using namespace babel;
+
+    stl::geometry geometry;
+
+    auto const data_as_string_view = cc::string_view(reinterpret_cast<char const*>(data.data()), reinterpret_cast<char const*>(data.end()));
+
+    auto const is_ascii = [&]() {
+        if (data.size() < 6)
+        {
+            on_error(data, data, "STL-file too short to be valid", severity::error);
+        }
+        if (!data_as_string_view.trim().starts_with("solid"))
+            return false;
+
+        // unfortunately many binary stl also start with "solid"
+        // also check for '\n' followed by "facet"
+        int idx = -1;
+        for (size_t i = 0; i < data_as_string_view.size(); ++i)
```

Comment:
when https://github.com/project-arcana/clean-core/pull/105 is merged we have a `index_of` in string_view

Reply:
Nice, that is a feature I was missing :+1:

Reply:
got merged :D

Reply:
Updated :+1: